packages/sdk/src/types.ts

```diff
@@ -60,41 +60,63 @@ export type MakeBuyOrderParams = MakeSellOrderParams;
 export enum MarketplaceName {
   Opensea = "opensea",
   Trader = "trader",
-  Looksrare = "looksrare",
+  LooksRare = "looksrare",
 }
 
-interface Opensea {
+interface ErrorObj {
+  message?: string;
+}
+
+interface ErrorBase {
+  error?: ErrorObj;
+}
+
+interface OpenseaBase extends ErrorBase {
   marketplaceName: MarketplaceName.Opensea;
 }
 
-interface Trader {
+interface TraderBase extends ErrorBase {
   marketplaceName: MarketplaceName.Trader;
 }
 
-interface Looksrare {
-  marketplaceName: MarketplaceName.Looksrare;
+interface LooksRareBase extends ErrorBase {
+  marketplaceName: MarketplaceName.LooksRare;
 }
 
-export interface OpenseaOrder extends Opensea {
-  marketplaceOrder: _OpenseaOrder;
+export interface NormalizedAsset {
+  contractAddress: string;
+  tokenId?: string;
+  type?: string;
+  amount: string;
 }
 
-export interface TraderOrder extends Trader {
-  marketplaceOrder: PostOrderResponsePayload;
+export interface OrderData<OriginalOrder> {
+  id: string;
+  makerAssets: NormalizedAsset[];
+  takerAssets: NormalizedAsset[];
+  isSellOrder: boolean;
```

Comment:
These 3 fields cannot come together become `makerAssets` already denotes the assets to be sold / offered.

If you want to use `isSellOrder` then you need to use `baseAssets` and `quoteAssets`. The order is always either buying or selling the base asset.

Alternatively you can use `assets` and `erc20Token` / `paymentToken` to be buying or selling the assets. But will be problematic if we want to trade nft for nft.

Reply:
This new structure is tied to new seaport orders.
You can see `OrderV2` type here - https://github.com/ProjectOpenSea/opensea-js/blob/master/src/orders/types.ts#L40
As you can see, both maker and taker assets are now can be bundles. Even right now, if you sell single nft for erc20, you are receiving 2 bundles with 1 item in each. So in a near future they might upgrade their backend to actually allow you to buy multiple nfts with multiple erc20 tokens.

But here is another dumb thing - in current version of opensea api, price for erc20 token is returned on root level of order in price field. But I don't have a way of reading minds of opensea devs, so there is a chance that they might allow some weird combos like [nft, nft] for [erc20, nft].

Reply:
I believe YJ's point on this is more on the terminology used e.g. in finance, `buy`/`sell` is not suitable to be used with `maker`/`taker`. So it's either:
bid/ask + maker/taker
OR
buy/sell + baseAsset/quoteAsset

I personally don't have much of an opinion on this but if we have to choose one or the other, sticking with maker/taker + bid/ask might be more suitable so that we dont introduce more terms i.e. baseAsset/quoteAsset

Feel free to hash this one out with YJ lol